src/simsopt/field/force.py

```diff
@@ -182,92 +442,2193 @@ def mean_squared_force_pure(gamma, gammadash, gammadashdash, quadpoints, current
+        (6 * a_list ** 2 * b_list ** 2) * jnp.log(a_list / b_list + b_list / a_list)
+    delta_ab = jnp.exp(-25.0 / 6.0 + k) * a_list * b_list
+    diag_rect = jnp.sum(jnp.sum(gammadash_prod / jnp.sqrt(rij_norm ** 2 + delta_ab[None, :, None, None]),
+                                axis=-1), axis=-1) / jnp.shape(gammas)[1] ** 2
+
+    # Use where to select diagonal elements based on cross-section type
+    diag_mask = jnp.eye(N, dtype=bool)
+    diag_values = jnp.where(cross_section == 'circular', diag_circ, diag_rect)
+
+    # Update diagonal elements
+    Lij = jnp.where(diag_mask, diag_values, Lij)
+    return 1e-7 * Lij
+
+
+def coil_coil_inductances_inv_pure(gammas, gammadashs, a_list, b_list, downsample, cross_section):
+    """
+    Pure function for computing the inverse of the coil inductance matrix.
+
+    Args:
+        gammas (array): Array of coil positions.
+        gammadashs (array): Array of coil tangent vectors.
+        a_list (array): Array of coil radii.
+        b_list (array): Array of coil thicknesses.
+        downsample (int): Downsample factor for the objective function.
+        cross_section (str): Cross-section type.
+
+    Returns:
+        array: Array of inverse of the coil inductance matrix.
+    """
+    # Lij is symmetric positive definite so has a cholesky decomposition
+    C = jnp.linalg.cholesky(coil_coil_inductances_full_pure(gammas, gammadashs, a_list, b_list, downsample, cross_section))
+    inv_C = jscp.linalg.solve_triangular(C, jnp.eye(C.shape[0]), lower=True)
+    inv_L = jscp.linalg.solve_triangular(C.T, inv_C, lower=False)
+    return inv_L
+
+
+def coil_currents_barebones(gammas, gammadashs, gammas_TF, gammadashs_TF, currents_TF, a_list, b_list, downsample, cross_section):
+    """
+    Pure function for computing the induced currents in a set of passive coils 
+    due to a set of TF coils + the other passive coils. 
+
+    Args:
+        gammas (array): Array of passive coil positions.
+        gammadashs (array): Array of passive coil tangent vectors.
+        gammas_TF (array): Array of TF coil positions.
+        gammadashs_TF (array): Array of TF coil tangent vectors.
+        currents_TF (array): Array of TF coil current.
+        a_list (array): Array of passive coil radii.
+        b_list (array): Array of passive coil thicknesses.
+        downsample (int): Downsample factor for the objective function.
+        cross_section (str): Cross-section type.
+
+    Returns:
+        array: Array of induced currents.
+    """
+    return -coil_coil_inductances_inv_pure(gammas, gammadashs, a_list, b_list, downsample, cross_section) @ net_fluxes_pure(gammas, gammadashs, gammas_TF, gammadashs_TF, currents_TF, downsample)
+
+
+def tve_pure(gamma, gammadash, gammas2, gammadashs2, current, currents2, a, b, downsample, cross_section):
+    r"""Pure function for minimizing the total vacuum energy on a coil.
+
+    The function is
+
+     .. math::
+        J = \frac{1}{2}I_iL_{ij}I_j
```

Comment:
This equation doesn't quite correspond to what this function actually does. The equation suggests - using the summation convention - that both of the {i,j} indices are summed over. Actually this function sums over one of the {i,j} indices but not over the other. The docstring should reflect this. Also add a space between `I_i` and `L` for legibility.

Reply:
fixing the docstring now